### PR TITLE
[FIX] Auto enabling anti cheat runtimes

### PR DESCRIPTION
**Pull request by Etaash-mathamsetty**
This has already been merged in the past, but got reverted in some refactoring

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: src/backend/storeManagers/legendary/games.ts [hunk 1]
```diff
@@ -637,12 +637,15 @@ export async function install(
   const anticheatInfo = gameAnticheatInfo(getGameInfo(appName).namespace)
 
   if (anticheatInfo && isLinux) {
-    const gameSettings = await getSettings(appName)
+    const gameConfig = GameConfig.get(appName)
+
+    if (anticheatInfo.anticheats.includes('Easy Anti-Cheat')) {
+      gameConfig.setSetting('eacRuntime', true)
+      if (isFlatpak) gameConfig.setSetting('useGameMode', true)
```
Comment (flavioislima) [inline src/backend/storeManagers/legendary/games.ts:644]: We had issues in the past with this approach, is safer to just use brackets like in other places.

File: src/backend/storeManagers/legendary/games.ts [hunk 2]
```diff
@@ -637,12 +637,15 @@ export async function install(
...
+    }
 
-    gameSettings.eacRuntime =
-      anticheatInfo.anticheats.includes('Easy Anti-Cheat')
-    if (gameSettings.eacRuntime && isFlatpak) gameSettings.useGameMode = true
-    gameSettings.battlEyeRuntime = anticheatInfo.anticheats.includes('BattlEye')
+    if (anticheatInfo.anticheats.includes('BattlEye'))
```
Comment (flavioislima) [inline src/backend/storeManagers/legendary/games.ts:647]: same here.

**Review by flavioislima — APPROVED** [review]
LGTM

**Comment by Etaash-mathamsetty** [comment]
now that I think about we should just have this on by default...

**Comment by gabriele2000** [comment]
> now that I think about we should just have these runtimes on by default... they can't cause crashes/issues as far as I know

I have, on `~/.profile`, two `export` lines that points to the EAC and BE runtimes, are you sure this "automatic" feature won't create conflicts like it did a year ago?

**Comment by imLinguin** [comment]
The variables will be shadowed (overwritten) when Heroic launches the game exe. With your configuration even if the option is disabled in Heroic, the anti cheats are enabled. 

I don't really understand why you do it like that, but I'm sure you have your reasons

**Comment by gabriele2000** [comment]
> I don't really understand why you do it like that, but I'm sure you have your reasons

There were some games on Steam that required manual anticheat runtime arguments and I eventually got tired of this.
Also, it was before Heroic introduced the toggle for EAC and BE, if I'm not mistaken.

What's the advantage of not having the runtime always loaded?

**Comment by Etaash-mathamsetty** [comment]
> > I don't really understand why you do it like that, but I'm sure you have your reasons
> 
> There were some games on Steam that required manual anticheat runtime arguments and I eventually got tired of this. Also, it was before Heroic introduced the toggle for EAC and BE, if I'm not mistaken.
> 
> What's the advantage of not having the runtime always loaded?

that is my point, there are no advantages, so we should have it loaded by default

**Pull request merged**